internal/api/grpc/settings/v2beta/integration_test/settings_test.go

```diff
@@ -52,16 +53,18 @@ func TestServer_GetSecuritySettings(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			resp, err := Client.GetSecuritySettings(tt.ctx, &settings.GetSecuritySettingsRequest{})
-			if tt.wantErr {
-				assert.Error(t, err)
-				return
-			}
-			require.NoError(t, err)
-			got, want := resp.GetSettings(), tt.want.GetSettings()
-			assert.Equal(t, want.GetEmbeddedIframe().GetEnabled(), got.GetEmbeddedIframe().GetEnabled(), "enable iframe embedding")
-			assert.Equal(t, want.GetEmbeddedIframe().GetAllowedOrigins(), got.GetEmbeddedIframe().GetAllowedOrigins(), "allowed origins")
-			assert.Equal(t, want.GetEnableImpersonation(), got.GetEnableImpersonation(), "enable impersonation")
+			assert.EventuallyWithT(t, func(ct *assert.CollectT) {
+				resp, err := Client.GetSecuritySettings(tt.ctx, &settings.GetSecuritySettingsRequest{})
+				if tt.wantErr {
+					assert.Error(ct, err)
+					return
+				}
+				require.NoError(ct, err)
```

Comment:
Require in an eventually test?

Reply:
Yes, because the request should never return an error, even if the projection isn't up-to-date yet.
If there is an error returned something else is wrong. Like lacking permissions or invalid token.
There is not need to wait (and waste time) for state to appear if there is an error here. 

So this panics the test instantly if there is an error.

Reply:
Works for me, was just stuck in the discussion we had to remove all require from the query tests